open_spiel/games/dou_dizhu/dou_dizhu_utils.cc

```diff
@@ -0,0 +1,840 @@
+// Copyright 2022 DeepMind Technologies Limited
```

Comment:
File level comment; can you make all variables constant that can be constant?

Can you also a) replace passing in containers by value with accepting const references (or span<const T>) and b) replace passing in containers by reference with accepting span<T>/span<const T>?

Reply:
Please see my response below.